### PR TITLE
opencv 4.x: fix cross-build to iOS/tvOS/watchOS + bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

fixes https://github.com/conan-io/conan-center-index/issues/6701

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

**Comment by SpaceIm** [comment]
It's worth noting that `WITH_JASPER` and `WITH_TIFF` are forced disabled by OpenCV CMakeLists if iOS. Maybe we should remove those options if iOS in another PR.

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -99,8 +111,11 @@ def configure(self):
         if not self.options.with_cuda:
             del self.options.with_cublas
             del self.options.with_cufft
-        self.options["libtiff"].jpeg = self.options.with_jpeg
-        self.options["jasper"].with_libjpeg = self.options.with_jpeg
+        if bool(self.options.with_jpeg):
+            if self.options.get_safe("with_jpeg2000") == "jasper":
+                self.options["jasper"].with_libjpeg = self.options.with_jpeg
+            if self.options.get_safe("with_tiff"):
+                self.options["libtiff"].jpeg = self.options.with_jpeg
```
Comment (SpaceIm) [inline recipes/opencv/4.x/conanfile.py:118]: previous logic here was broken. We want to use the same jpeg backend in all dependencies, not to disable jpeg support in dependencies when jpeg is disabled in opencv.

**Review by uilianries — APPROVED** [review]

File: recipes/opencv/4.x/conanfile.py (hunk 1 above)
Comment (jgsogo) [reply recipes/opencv/4.x/conanfile.py:118]: I tipically prefer to raise if values mismatch than to force a value for a dependency, but both approaches will work almost the same. 

Anyway, the `provides` mechanism should prevent mixing different `jpeg` implementations in the same graph (for the good and the bad), if recipes are declaring it.

**Review by jgsogo — APPROVED** [review]

**Comment by SpaceIm** [comment]
@jgsogo I understand, I didn't introduce this mechanism in this PR, it was already there but potentially broken.

File: recipes/opencv/4.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py:118]: I understand, I didn't introduce this mechanism in this PR, it was already there but potentially broken.

Comment (SSE4) [reply recipes/opencv/4.x/conanfile.py:118]: yes, eventually we will be able to remove this code altogether

**Review by SSE4 — APPROVED** [review]

**Pull request merged**